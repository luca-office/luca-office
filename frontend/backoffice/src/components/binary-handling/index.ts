export * from "./binary-form-placeholder/binary-form-placeholder"
export * from "./binary-update-modal/binary-entry"
export * from "./binary-update-modal/binary-update-modal"
export * from "./detail-view-binary/detail-view-binary"
export * from "./image-upload-modal/image-upload-modal"
