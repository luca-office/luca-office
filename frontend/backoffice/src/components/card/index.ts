// special backoffice card elements
export * from "./creation-card"
export * from "./duration-info/card-duration-info"
export * from "./tabbed-card/__mocks__/card-tabs.mock"
export * from "./tabbed-card/tabbed-card"
