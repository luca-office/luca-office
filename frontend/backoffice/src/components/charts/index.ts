export * from "./donut-chart/donut-chart-with-legend"
