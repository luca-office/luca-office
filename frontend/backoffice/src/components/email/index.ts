export * from "./email-body-meta-entry/email-body-meta-entry"
export * from "./email-body-setting-selection/email-body-setting-selection"
export * from "./email-content/email-content"
export * from "./email-directories-list/email-directories-list"
export * from "./email-header/email-header"
export * from "./email-unread-indicator/email-unread-indicator"
