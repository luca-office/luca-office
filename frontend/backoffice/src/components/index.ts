export * from "./binary-handling"
export * from "./card"
export * from "./card-overview/card-overview"
export * from "./charts"
export * from "./creation-bar/creation-bar"
export * from "./creation-status/creation-status"
export * from "./editable-multiline-textarea/editable-multiline-textarea"
export * from "./editing-status-indicator/editing-status-indicator"
export * from "./email"
export * from "./information-entry/information-entry"
export * from "./inline-editable-header/inline-editable-header"
export * from "./inline-editable-textarea/inline-editable-textarea"
export * from "./markdown-text/markdown-text"
export * from "./markdown-text-field/markdown-text-field"
export * from "./markdown-text-update-modal/markdown-text-update-modal"
export * from "./meta-entry/meta-entry"
export * from "./meta-entry-tags/meta-entry-tags"
export * from "./overlay-edit-field/overlay-edit-field"
export * from "./rating-progress-card/rating-progress-card"
export * from "./rating-progress-card/utils"
export * from "./resort-modal/resort-modal"
export * from "./sub-header-detail/sub-header-detail"
export * from "./sub-header-detail/sub-header-detail-container"
export * from "./sub-header-filter/sub-header-filter"
export * from "./table-action-bar/table-action-bar"
export * from "./upload-file-modal/upload-file-modal"
