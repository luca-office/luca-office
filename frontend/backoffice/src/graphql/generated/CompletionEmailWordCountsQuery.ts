/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: CompletionEmailWordCountsQuery
// ====================================================

export interface CompletionEmailWordCountsQuery_completionEmailWordCounts {
  __typename: "CompletionEmailWordCount";
  invitationId: string;
  wordCount: number | null;
}

export interface CompletionEmailWordCountsQuery {
  completionEmailWordCounts: CompletionEmailWordCountsQuery_completionEmailWordCounts[];
}

export interface CompletionEmailWordCountsQueryVariables {
  surveyId: string;
  scenarioId: string;
}
