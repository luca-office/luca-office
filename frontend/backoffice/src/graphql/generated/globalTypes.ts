/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum MimeType {
  ApplicationPdf = "ApplicationPdf",
  ImageGif = "ImageGif",
  ImageJpeg = "ImageJpeg",
  ImagePng = "ImagePng",
  ImageSvg = "ImageSvg",
  Spreadsheet = "Spreadsheet",
  TextHtml = "TextHtml",
  VideoMp4 = "VideoMp4",
}

export enum Relevance {
  Irrelevant = "Irrelevant",
  PotentiallyHelpful = "PotentiallyHelpful",
  Required = "Required",
}

export interface DirectoryCreation {
  name: string;
  parentDirectoryId?: string | null;
  scenarioId?: string | null;
  sampleCompanyId?: string | null;
}

export interface DirectoryUpdate {
  name: string;
  parentDirectoryId?: string | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
