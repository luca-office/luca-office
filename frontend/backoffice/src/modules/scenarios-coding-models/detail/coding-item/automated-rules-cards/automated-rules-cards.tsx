import {css} from "@emotion/react"
import * as React from "react"
import {Column, Columns, Label, SelectableCard, SelectionIconType} from "shared/components"
import {AutomatedCodingItemRule} from "shared/graphql/generated/globalTypes"
import {spacingHuge, spacingMedium} from "shared/styles"
import {useLucaTranslation} from "shared/translations"

export interface AutomatedRulesCardsProps {
  readonly selectedRule: AutomatedCodingItemRule
  readonly setSelectedRule: React.Dispatch<React.SetStateAction<AutomatedCodingItemRule>>
}

export const AutomatedRulesCards: React.FC<AutomatedRulesCardsProps> = ({selectedRule, setSelectedRule}) => {
  const {t} = useLucaTranslation()

  return (
    <>
      <Label label={t("coding_models__create_item_label_rating_type_automated_title")} />
      <Columns>
        <Column>
          <SelectableCard
            customStyles={styles.typeCard}
            selectionIconType={SelectionIconType.RADIO}
            selected={selectedRule === AutomatedCodingItemRule.DocumentView}
            onClick={() => setSelectedRule(AutomatedCodingItemRule.DocumentView)}
            text={t("coding_models__create_item_label_rating_type_automated_document_view_hint")}
            title={t("coding_models__create_item_label_rating_type_automated_document_view")}
          />
        </Column>
        <Column>
          <SelectableCard
            customStyles={styles.typeCard}
            selectionIconType={SelectionIconType.RADIO}
            selected={selectedRule === AutomatedCodingItemRule.FeatureUsage}
            onClick={() => setSelectedRule(AutomatedCodingItemRule.FeatureUsage)}
            text={t("coding_models__create_item_label_rating_type_automated_feature_usage_hint")}
            title={t("coding_models__create_item_label_rating_type_automated_feature_usage")}
          />
        </Column>
        <Column>
          <SelectableCard
            customStyles={styles.typeCard}
            selectionIconType={SelectionIconType.RADIO}
            selected={selectedRule === AutomatedCodingItemRule.ToolUsage}
            onClick={() => setSelectedRule(AutomatedCodingItemRule.ToolUsage)}
            text={t("coding_models__create_item_label_rating_type_automated_tool_usage_hint")}
            title={t("coding_models__create_item_label_rating_type_automated_tool_usage")}
          />
        </Column>
      </Columns>
      <Columns customStyles={styles.secondRowWrapper}>
        <Column customStyles={styles.secondRow}>
          <SelectableCard
            customStyles={styles.typeCard}
            selectionIconType={SelectionIconType.RADIO}
            selected={selectedRule === AutomatedCodingItemRule.InputValue}
            onClick={() => setSelectedRule(AutomatedCodingItemRule.InputValue)}
            text={t("coding_models__create_item_label_rating_type_automated_input_value_hint")}
            title={t("coding_models__create_item_label_rating_type_automated_input_value")}
          />
        </Column>
        <Column customStyles={styles.secondRow}>
          <SelectableCard
            customStyles={styles.typeCard}
            selectionIconType={SelectionIconType.RADIO}
            selected={selectedRule === AutomatedCodingItemRule.RScript}
            onClick={() => setSelectedRule(AutomatedCodingItemRule.RScript)}
            text={t("coding_models__create_item_label_rating_type_automated_r_script_hint")}
            title={t("coding_models__create_item_label_rating_type_automated_r_script")}
          />
        </Column>
      </Columns>
    </>
  )
}

const styles = {
  hint: css({
    marginBottom: spacingMedium
  }),
  row: css({
    marginTop: spacingHuge
  }),
  secondRow: css({
    flexGrow: 0,
    flexBasis: "33%"
  }),
  secondRowWrapper: css({
    marginTop: spacingMedium
  }),
  typeCard: css({
    height: 100
  })
}
