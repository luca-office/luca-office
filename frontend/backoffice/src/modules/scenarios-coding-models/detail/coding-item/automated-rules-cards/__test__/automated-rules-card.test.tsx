import {shallow} from "enzyme"
import React from "react"
import {create} from "react-test-renderer"
import {Column, Columns, Label, SelectableCard} from "shared/components"
import {AutomatedCodingItemRule} from "shared/graphql/generated/globalTypes"
import {AutomatedRulesCards, AutomatedRulesCardsProps} from "../automated-rules-cards"

const defaultProps: AutomatedRulesCardsProps = {
  selectedRule: AutomatedCodingItemRule.DocumentView,
  setSelectedRule: jest.fn()
}

const getComponent = (props?: Partial<AutomatedRulesCardsProps>) => (
  <AutomatedRulesCards {...{...defaultProps, ...props}} />
)

describe("AutomatedRulesCards", () => {
  it("renders correctly with default props", () => {
    const component = getComponent()
    const tree = create(component).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("has correct structure (default props)", () => {
    const component = shallow(getComponent())

    expect(component.find(Label)).toHaveLength(1)
    expect(component.find(Columns)).toHaveLength(2)
    expect(component.find(Column)).toHaveLength(5)
    expect(component.find(SelectableCard)).toHaveLength(5)
  })
})
