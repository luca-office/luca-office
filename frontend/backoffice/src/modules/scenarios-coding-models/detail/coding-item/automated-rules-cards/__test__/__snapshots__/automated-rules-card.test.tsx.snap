// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutomatedRulesCards renders correctly with default props 1`] = `
[
  <div
    className="css-lz4b3k-Label"
  >
    coding_models__create_item_label_rating_type_automated_title
  </div>,
  <div
    className="css-1cky3ub-Columns"
  >
    <div
      className="css-1oiyj7r-Column"
    >
      <div
        className="css-6egt7t-Card"
        onClick={[Function]}
      >
        <div
          className="card-header css-a0e080-CardHeader"
        >
          <div
            className="css-1cygpbo-SelectableCard"
          >
            <div
              className="css-17tkprm-SelectableCard"
            >
              <div
                className="css-1f6p2vj-Heading"
              >
                coding_models__create_item_label_rating_type_automated_document_view
              </div>
            </div>
            <div
              className="radio-button css-aqrmqg-RadioButton"
            >
              <div
                className="css-1dzk059-RadioButton"
              >
                <div
                  className="css-1kv9hnx-RadioButton"
                >
                  <div
                    className="css-1lbkotz-RadioButton"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-hpovgn-Text"
        >
          coding_models__create_item_label_rating_type_automated_document_view_hint
        </div>
      </div>
    </div>
    <div
      className="css-1oiyj7r-Column"
    >
      <div
        className="css-1h76oty-Card"
        onClick={[Function]}
      >
        <div
          className="card-header css-a0e080-CardHeader"
        >
          <div
            className="css-1cygpbo-SelectableCard"
          >
            <div
              className="css-17tkprm-SelectableCard"
            >
              <div
                className="css-1f6p2vj-Heading"
              >
                coding_models__create_item_label_rating_type_automated_feature_usage
              </div>
            </div>
            <div
              className="radio-button css-aqrmqg-RadioButton"
            >
              <div
                className="css-1dzk059-RadioButton"
              >
                <div
                  className="css-1kv9hnx-RadioButton"
                >
                  <div
                    className="css-7sxxnu-RadioButton"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-hpovgn-Text"
        >
          coding_models__create_item_label_rating_type_automated_feature_usage_hint
        </div>
      </div>
    </div>
    <div
      className="css-1oiyj7r-Column"
    >
      <div
        className="css-1h76oty-Card"
        onClick={[Function]}
      >
        <div
          className="card-header css-a0e080-CardHeader"
        >
          <div
            className="css-1cygpbo-SelectableCard"
          >
            <div
              className="css-17tkprm-SelectableCard"
            >
              <div
                className="css-1f6p2vj-Heading"
              >
                coding_models__create_item_label_rating_type_automated_tool_usage
              </div>
            </div>
            <div
              className="radio-button css-aqrmqg-RadioButton"
            >
              <div
                className="css-1dzk059-RadioButton"
              >
                <div
                  className="css-1kv9hnx-RadioButton"
                >
                  <div
                    className="css-7sxxnu-RadioButton"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-hpovgn-Text"
        >
          coding_models__create_item_label_rating_type_automated_tool_usage_hint
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-193kak8-Columns"
  >
    <div
      className="css-s1bg9q-Column"
    >
      <div
        className="css-1h76oty-Card"
        onClick={[Function]}
      >
        <div
          className="card-header css-a0e080-CardHeader"
        >
          <div
            className="css-1cygpbo-SelectableCard"
          >
            <div
              className="css-17tkprm-SelectableCard"
            >
              <div
                className="css-1f6p2vj-Heading"
              >
                coding_models__create_item_label_rating_type_automated_input_value
              </div>
            </div>
            <div
              className="radio-button css-aqrmqg-RadioButton"
            >
              <div
                className="css-1dzk059-RadioButton"
              >
                <div
                  className="css-1kv9hnx-RadioButton"
                >
                  <div
                    className="css-7sxxnu-RadioButton"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-hpovgn-Text"
        >
          coding_models__create_item_label_rating_type_automated_input_value_hint
        </div>
      </div>
    </div>
    <div
      className="css-s1bg9q-Column"
    >
      <div
        className="css-1h76oty-Card"
        onClick={[Function]}
      >
        <div
          className="card-header css-a0e080-CardHeader"
        >
          <div
            className="css-1cygpbo-SelectableCard"
          >
            <div
              className="css-17tkprm-SelectableCard"
            >
              <div
                className="css-1f6p2vj-Heading"
              >
                coding_models__create_item_label_rating_type_automated_r_script
              </div>
            </div>
            <div
              className="radio-button css-aqrmqg-RadioButton"
            >
              <div
                className="css-1dzk059-RadioButton"
              >
                <div
                  className="css-1kv9hnx-RadioButton"
                >
                  <div
                    className="css-7sxxnu-RadioButton"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-hpovgn-Text"
        >
          coding_models__create_item_label_rating_type_automated_r_script_hint
        </div>
      </div>
    </div>
  </div>,
]
`;
